src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -725,16 +724,6 @@ private function getOrderDiscounts(Order $order): OrderDiscountsForViewing
 
         foreach ($discounts as $discount) {
             $discountAmount = $isTaxIncluded ? $discount['value'] : $discount['value_tax_excl'];
-
-            $cartRule = new CartRule((int) $discount['id_cart_rule']);
```

Comment:
We dont need this because the stored discount is already converted to order currency

Reply:
It is true for discount that were manually added to the order via the BO because they are created with the matching currency However a CartRule can be define as a generic shop cart rule in which case it will probably we configured with the default shop currency which is not necessarily the Order's one

Reply:
I just tested the scenario. when the cartRule is applied on FO or when the product is added through the BO, the cartRule discount is converted when it is applied

Reply:
mmm, so this means a generic CartRule is copied for each cart created from the FO?

Reply:
Oh I understand, thats because the initial call is `$order->getCartRules()` which actually get OrderCartRules list, these one are indeed created for each order creation which contextual values